autodocodec/src/Autodocodec/Class.hs

```diff
@@ -144,31 +165,29 @@ instance (Ord a, HasCodec a) => HasCodec (Set a) where
   codec = dimapCodec S.fromList S.toList codec
 
 instance (Ord k, FromJSONKey k, ToJSONKey k, HasCodec v) => HasCodec (Map k v) where
-  codec = MapCodec codec
+  codec = mapCodec codec
 
 instance (Eq k, Hashable k, FromJSONKey k, ToJSONKey k, HasCodec v) => HasCodec (HashMap k v) where
-  codec = HashMapCodec codec
+  codec = hashMapCodec codec
 
 #if MIN_VERSION_aeson(2,0,0)
 instance HasCodec v => HasCodec (KeyMap v) where
   codec = keyMapCodec codec
 #endif
-
--- TODO make these instances better once aeson exposes its @Data.Aeson.Parser.Time@ or @Data.Attoparsec.Time@ modules.
 instance HasCodec Day where
-  codec = codecViaAeson "Day"
+  codec = codecViaISO8601 "Day"
```

Comment:
I wonder if the roundtrip tests with aeson start failing after this change.

Reply:
I haven't noticed this causing issues in the test runs.

Reply:
It seems to work currently, but I see the point that we want to keep consistency with aeson. 

I've instead changed the approach as a result. I think I really should put a PR to aeson, but in the meantime I think the solution I've put now is better, even if it's a bit scary.